### PR TITLE
TPC: Update TPC Trending 

### PR DESCRIPTION
**Pull request by mlesch**
Changes: 

- Add possibility to trend vs run-number 
- Add possibility to add a new histogram as datasource for a continues trend

### CODE REVIEW

File: Modules/TPC/src/TrendingTaskTPC.cxx [hunk 1]
```diff
@@ -89,12 +89,21 @@ void TrendingTaskTPC::initialize(Trigger, framework::ServiceRegistryRef services
     mTrend->SetBranchAddress("meta", &mMetaData);
     mTrend->SetBranchAddress("time", &mTime);
     for (const auto& source : mConfig.dataSources) {
+      bool existingBranch = mTrend->GetBranchStatus(source.name.c_str());
```
Comment (wiechula) [inline Modules/TPC/src/TrendingTaskTPC.cxx:92]:
```suggestion
      const bool existingBranch = mTrend->GetBranchStatus(source.name.c_str());
```

File: Modules/TPC/src/TrendingTaskTPC.cxx [hunk 2]
```diff
@@ -289,22 +298,30 @@ void TrendingTaskTPC::drawCanvasMO(TCanvas* thisCanvas, const std::string& var,
...
       int iEntry = 0;
 
-      graphErrors = new TGraphErrors(nEntries);
+      graphErrors = new TGraphErrors(nEffectiveEntries);
+      myReader.SetEntry(startPoint - 1); // startPoint-1 as myReader.Next() increments by one so that we then start at startPoint
 
       while (myReader.Next()) {
-        const double timeStamp = (double)(*retrieveTime);
+        const double timeStamp = (trendType == "time") ? (double)(*retrieveTime) : (double)(*retrieveRun);
+        ;
```
Comment (wiechula) [inline Modules/TPC/src/TrendingTaskTPC.cxx:324]: There is an empty line with `;`. And perhaps for better readability `timeStamp` -> `xVal`
```suggestion
        const double xVal = (trendType == "time") ? (double)(*retrieveTime) : (double)(*retrieveRun);

```

File: Modules/TPC/src/TrendingTaskTPC.cxx [hunk 3]
```diff
@@ -332,17 +349,22 @@ void TrendingTaskTPC::drawCanvasMO(TCanvas* thisCanvas, const std::string& var,
...
     for (int p = 0; p < nuPa; p++) {
       int iEntry = 0;
-      auto gr = new TGraphErrors(nEntries);
+      auto gr = new TGraphErrors(nEffectiveEntries);
+      myReader.SetEntry(startPoint - 1); // startPoint-1 as myReader.Next() increments by one so that we then start at startPoint
 
       while (myReader.Next()) {
-        const double timeStamp = (double)(*retrieveTime);
+        const double timeStamp = (trendType == "multigraphtime") ? (double)(*retrieveTime) : (double)(*retrieveRun);
+        ;
```
Comment (wiechula) [inline Modules/TPC/src/TrendingTaskTPC.cxx:367]: See above
```suggestion
        const double xVal = (trendType == "multigraphtime") ? (double)(*retrieveTime) : (double)(*retrieveRun);
        
```

File: Modules/TPC/src/TrendingTaskTPC.cxx [hunk 4]
```diff
@@ -493,21 +515,29 @@ void TrendingTaskTPC::drawCanvasQO(TCanvas* thisCanvas, const std::string& var,
...
+  const int startPoint = (trendType == "time") ? nEntriesTime - nEffectiveEntries : nEntriesRuns - nEffectiveEntries;
+
   int iEntry = 0;
-  graphErrors = new TGraphErrors(nEntries);
+  graphErrors = new TGraphErrors(nEffectiveEntries);
+  myReader.SetEntry(startPoint - 1); // startPoint-1 as myReader.Next() increments by one so that we then start at startPoint
 
   while (myReader.Next()) {
-    const double timeStamp = (double)(*retrieveTime);
+    const double timeStamp = (trendType == "time") ? (double)(*retrieveTime) : (double)(*retrieveRun);
```
Comment (wiechula) [inline Modules/TPC/src/TrendingTaskTPC.cxx:540]: See above
```suggestion
    const double xVal = (trendType == "time") ? (double)(*retrieveTime) : (double)(*retrieveRun);
```

**Review by wiechula — CHANGES_REQUESTED** [review]
Just minor comments

**Comment by mlesch** [comment]
Thanks! I implemented all comments as suggested

**Review by wiechula — APPROVED** [review]

**Pull request merged**